### PR TITLE
extend the journald.conf template to cover all parameters

### LINKED ISSUE #2: RFE:  Complete the template
Hi again Christian,
Currently I use this non-galaxy role, which is a bit more complete but appears to be abandoned (AFAICT, the sole Ansible dev w/ Stone Payments works elsewhere now):  https://github.com/stone-payments/ansible-journald

As you can see from my PR over there [#2](https://github.com/stone-payments/ansible-journald/pull/2), I'm not having much luck, and also it is not as nice to use that Github based role as it would be to use a Galaxy Role that can be installed and managed using the normal methods.

As such, what do you think about using that role as inspiration to make your role more feature-complete?  I suspect that this role will work just fine with most systemd distros, including Ubuntu 18.04+ and RHEL/CentOS 7+

### PR DESCRIPTION
**Pull request by kfiresmith**
journald.conf.j2 now includes an optional role variable for each parameter
included with journald.  Additionally, I have added the configuration
file preamble included with the latest upstream journald release.

The templated file also now includes the {{ ansible_managed }}
standard heading.

Lastly, the README for the role was updated to cover this change.

fixes #2

### CODE REVIEW

**Comment by kfiresmith** [comment]
Hi Christian,
Hopefully this PR looks a bit better than the one for timesyncd.  I read through the contribution guide and style guide, and believe I've gotten most of this correct, but please let me know if there are further areas where I can improve the quality of my PRs for your projects.

I found the overall git-knowledge-base page quite useful.  The only place I wasn't certain after reading was how to name my branch, so as you can see I named it 'extend-template'.  In hindsight, perhaps `topic/name` in the contribution guide would map to something more like `enhancement/extend-template`, or `bug/fix-typo`?

**Comment by wookietreiber** [comment]
> I found the overall git-knowledge-base page quite useful. The only place I wasn't certain after reading was how to name my branch, so as you can see I named it 'extend-template'. In hindsight, perhaps topic/name in the contribution guide would map to something more like enhancement/extend-template, or bug/fix-typo?

Glad to hear you took a deeper dive into the knowledge base pages and that it's useful :smiley: 

Branch names aren't all too important if there aren't many of them and/or if they're short lived. Since this particular branch is only visible in your fork of the repository anyway, it's even less important. A good branch naming scheme, e.g. with the `topic/` prefix, can be extremely useful for truly shared and busy repositories, to which multiple people have push access. 

What's more important is preparing pull requests in a dedicated branch, so it's both easier to prepare multiple PRs at the same time and it's also easier to rebase if needed.

File: README.md [hunk 1]
```diff
@@ -30,8 +30,14 @@ Requirements
 Role Variables
 --------------
 
+Role variables include all supported journald parameters as documented in detail at the [freedesktop.org website](https://www.freedesktop.org/software/systemd/man/journald.conf.html#).
```
Comment (wookietreiber) [inline README.md]: Please put the link into the link collection at the bottom of the README (consistency).

File: README.md [hunk 2]
```diff
@@ -30,8 +30,14 @@ Requirements
...
+
+Each variable has the `systemd_journald_` prefix, and is named for the parameter.  For example, the `MaxRetentionSec=` parameter maps to the `systemd_journald_maxretensionsec` role variable.
+
+Because every parameter has a default, journald.conf starts completely commented out, until you choose to override a parameter default.
```
Comment (wookietreiber) [inline README.md]: Please clarify that the defaults are from **journald**, not the **role**.

File: README.md (hunk 1 above)
Comment (wookietreiber) [inline README.md]: Please hard wrap all lines to 79 characters (consistency).

File: README.md (hunk 2 above)
Comment (wookietreiber) [inline README.md]: Please put the code quotes around `journald.conf` (highlighting).

**Review by wookietreiber — CHANGES_REQUESTED** [review]
Mostly nitpicking. I have a certain style with my Markdown documents I haven't documented yet. Would be nice to have a review tool that does this automatically :smile:

**Comment by wookietreiber** [comment]
>  Additionally, I have added the configuration file preamble included with the latest upstream journald release.

That's awesome, I always do this, too, when I update a template. :+1:

**Comment by kfiresmith** [comment]
Thanks for the feedback.  I've attempted to try to fix this PR/branch first before dealing with the other simple PR to update distro support.  I've gotten myself into a git mess and am possibly in over my head trying to update the PR.  
I tried firing up GitKraken to see if I could visually make sense of the mess I'm in, but that's somehow even worse.  Any advice on this?
image

**Comment by wookietreiber** [comment]
This is the classic **merge vs. rebase** or **linear vs. non-linear history** discussion. I assume that at some point you did a `git merge upstream/master` instead of `git rebase upstream/master`. I've written [a bit in the knowledge base about that](https://idiv-biodiversity.github.io/git-knowledge-base/master-collab.html), and I prefer rebase'ing over merging to create a [linear history](https://idiv-biodiversity.github.io/git-knowledge-base/linear-vs-nonlinear.html) in all my projects. 

You can fix this with rebase'ing on top off the current master. From your local repository, it should go like this: 

```bash
# for completeness
git remote add upstream https://github.com/idiv-biodiversity/ansible-role-systemd-journald.git

# perform the rebase 
git fetch upstream
git rebase upstream/master extend-template
git push --force-with-lease
```

I've tested this locally and the rebase had no conflicts. The force push should update the PR.

**Comment by kfiresmith** [comment]
Hi @wookietreiber  - sadly this didn't work out for me, I think I'll have to start over.  I still have a detached head that seems to be marooned on it's own.

**Comment by kfiresmith** [comment]
OK, bailed on the last detached head and tried again, and I see an updated commit now in this PR so please let me know if all conditions/changes are met at this point.  Thanks.

**Comment by wookietreiber** [comment]
Sorry about the delay. I merged on the CLI, so will close here. I already cut a v1.0.0 release since we now support all parameters, thanks a lot for the contribution!

**Pull request closed without merging**